src/sif/autocost.cc

```diff
@@ -537,7 +543,17 @@ Cost AutoCost::TransitionCost(const baldr::DirectedEdge* edge,
                       ? kRightSideTurnCosts[static_cast<uint32_t>(edge->turntype(idx))]
                       : kLeftSideTurnCosts[static_cast<uint32_t>(edge->turntype(idx))];
     }
-    seconds += trans_density_factor_[node->density()] * edge->stopimpact(idx) * turn_cost;
+    turn_cost *= edge->stopimpact(idx);
+
+    // Separate time and penalty if using predicted speeds. Even when we reduce time
+    // for a particular turn we still may want to have slight penalties to help avoid
+    // unfavorable turns).
+    if (has_traffic) {
+      seconds += kTrafficTransitionFactor * turn_cost;
+      penalty += (trans_density_factor_[node->density()] - kTrafficTransitionFactor) * turn_cost;
+    } else {
+      seconds += trans_density_factor_[node->density()] * turn_cost;
+    }
```

Comment:
@dnesbitt61 are we going to add this to the other vehicular TransitionCost functions?

Reply:
Later, we need to consider how traffic should influence truck routing (may want to cap speeds at truck speed limit or slightly over that?) and even motorcycle routing (some places can split lanes)